cli/src/program.rs

```diff
@@ -2791,14 +2815,15 @@ fn send_deploy_messages(
                         solana_client::tpu_client::TpuClientConfig::default(),
                         cache,
                     );
-                    let tpu_client = rpc_client
+                    let tpu_client = use_rpc.then(|| rpc_client
                         .runtime()
                         .block_on(tpu_client_fut)
-                        .expect("Should return a valid tpu client");
+                        .expect("Should return a valid tpu client")
+                    );
 
                     send_and_confirm_transactions_in_parallel_blocking(
                         rpc_client.clone(),
-                        Some(tpu_client),
+                        tpu_client,
```

Comment:
this is the guts, right?

Reply:
Yeah exactly, `send_and_confirm_transactions_in_parallel` takes an optional TPU client. If it's present it always uses it, and if not, it uses RPC.

Reply:
this seems counterintuitive because i read this PR as `add --use-rpc so that your deploys will go through the RPC client not the TPU client` but its actually vice versa. when i did this https://github.com/staratlasmeta/solana/commit/eb6be985ec0fad8bee564aa4bb0c1c076a2fa7fc and rebuilt  as per https://twitter.com/stegaBOB/status/1775287709663109195?t=YQQQvJOgrOGnY-E5SCwfFg&s=19 my deploys are going through now

Reply:
Oh darn, I did get it backwards, thanks a bunch for noticing this! Fixing with #1008